### PR TITLE
fix(DB/gameobject): Fix overlapping Water Barrel and Food Crate in Kharanos

### LINKED ISSUE #6346: Dun Morogh - Water Barrel and Food Crates Overlapping in Kharanos
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [X] enUS
- [ ] other (specify)

##### FACTION

- [X] Alliance
- [ ] Horde

##### CONTENT PHASE:

- [ ] Generic
- [X] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:
Original report: https://github.com/chromiecraft/chromiecraft/issues/860

Gobs
1462 (Entry: 3658) Water Barrel 
10677 (Entry: 3719) Food Crate
have spawned, and continue to spawn together at the same spot (X:-5569.669922 Y:-462.497009 Z:404.397003 MapId:0)

Tester: confirmed: 
WoWScrnShot_061421_155019

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Only one of these gobs should be spawned at a time.

##### SOURCE:

Lack of comments on wowhead/wowpedia remarking an special spawning. Same objects on other zones spawn alone. 

##### STEPS TO REPRODUCE THE PROBLEM:

1. .go o 1462
2. Click on Food Crate and Water Barrel
3. Wait
4. Food Crate and Water Barrel spawn together again after some minutes.

##### EXTRA NOTES:
https://wowpedia.fandom.com/wiki/Water_Barrel
https://wowpedia.fandom.com/wiki/Food_Crate
https://www.wowhead.com/object=3658/water-barrel
https://www.wowhead.com/object=3662/food-crate

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/d0a52896283a9bfed36ca897f0c2ebe870a25800

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

### PR DESCRIPTION
**Pull request by mserajnik**
## Changes Proposed:

This puts both gameobjects into a pool where only one can spawn at a time and gives them an equal spawn chance.

## Issues Addressed:

- Closes #6346
- Closes https://github.com/chromiecraft/chromiecraft/issues/860

## SOURCE:

According to #6346:

> Lack of comments on wowhead/wowpedia remarking an special spawning. Same objects on other zones spawn alone.

## Tests Performed:

See _How to Test the Changes_.

## How to Test the Changes:

1. `.go o 1462`
2. Observe that either the Water Barrel or Food crate is spawned, but not both at the same time
3. Loot the Water Barrel or Food Crate
4. Wait a few minutes for the respawn and observe that either the Water Barrel or Food crate gets spawned, but not both at the same time

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by mserajnik** [comment]
From what I can see, are 5 other locations where Water Barrels and Food Crates spawn at the same time.

First, I simply figured out all gameobjects that share the same `position_x` and are of type Water Barrel (`3658`) or Food Crate (`3719`):

```sql
mysql> SELECT `position_x`, COUNT(*) AS `count` FROM `gameobject` WHERE `id` IN (3658, 3719) GROUP BY `position_x` HAVING `count` > 1 ORDER BY `position_x`;
+------------+-------+
| position_x | count |
+------------+-------+
|   -9729.42 |     2 |
|   -9500.53 |     2 |
|   -9458.53 |     2 |
|   -9258.88 |     2 |
|   -5569.67 |     2 |
|   -5470.22 |     2 |
+------------+-------+
6 rows in set (0.26 sec)
```

`position_x` `-5569.67` is the location of the two gameobjects fixed in this PR, so these can be ignored here.

I've checked all the other objects and confirmed that besides the same `position_x`, they also have the same `position_y`, `position_z` and `spawntimesecs` without being assigned to a pool, so they will exhibit the same issue as the two gameobjects fixed in this PR:

| guid  | id   | position_x | position_y | position_z | spawntimesecs |
|-------|------|------------|------------|------------|---------------|
| 29275 | 3658 | -9729.42   | 656.299    | 31.8338    | 180           |
| 30758 | 3719 | -9729.42   | 656.299    | 31.8338    | 180           |

| guid  | id   | position_x | position_y | position_z | spawntimesecs |
|-------|------|------------|------------|------------|---------------|
| 32313 | 3658 | -9500.53   | 81.368     | 57.4601    | 180           |
| 32754 | 3719 | -9500.53   | 81.368     | 57.4601    | 180           |

| guid  | id   | position_x | position_y | position_z | spawntimesecs |
|-------|------|------------|------------|------------|---------------|
| 30677 | 3658 | -9458.53   | 8.58518    | 56.9626    | 180           |
| 30839 | 3719 | -9458.53   | 8.58518    | 56.9626    | 180           |

| guid  | id   | position_x | position_y | position_z | spawntimesecs |
|-------|------|------------|------------|------------|---------------|
| 29276 | 3658 | -9258.88   | 355.379    | 76.693     | 180           |
| 30687 | 3719 | -9258.88   | 355.379    | 76.693     | 180           |

| guid  | id   | position_x | position_y | position_z | spawntimesecs |
|-------|------|------------|------------|------------|---------------|
| 29306 | 3719 | -5470.22   | -636.681   | 393.521    | 180           |
| 29609 | 3658 | -5470.22   | -636.681   | 393.521    | 180           |

I am not sure if the way I have fixed the gameobjects in this PR is good/acceptable, but if it is, I can make another PR that fixes these locations too after this one gets approved.

File: data/sql/updates/pending_db_world/rev_1623792334812044570.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1623792334812044570');
+
+SET @POOL = 60004;
+
+INSERT INTO `pool_template` (`entry`, `max_limit`, `description`) VALUES (@POOL, 1, 'Water Barrel (1462) / Food Crate (10677) - Kharanos');
+
+INSERT INTO `pool_gameobject` (`guid`, `pool_entry`, `chance`, `description`) VALUES
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1623792334812044570.sql:7]: please use `DELETE` before `INSERT`

https://www.azerothcore.org/wiki/sql-standards

Comment (mserajnik) [reply data/sql/updates/pending_db_world/rev_1623792334812044570.sql:7]: Sorry, I missed that somehow. Will fix.

**Comment by claudiodfc** [comment]
If every other gobject is fixed like this lets keep consistency. At least in the future we know where to improve! 👍🏼

File: data/sql/updates/pending_db_world/rev_1623792334812044570.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1623792334812044570');
+
+SET @POOL = 60004;
```
Comment (claudiodfc) [inline data/sql/updates/pending_db_world/rev_1623792334812044570.sql:3]: I'm gonna start using these set variables for my queries aswell!

Comment (mserajnik) [reply data/sql/updates/pending_db_world/rev_1623792334812044570.sql:3]: @claudiodfc

I took inspiration from PRs like #5071.

I like the idea of not repeating values if they are used multiple times; makes it easier to read and avoids typos. :D

Comment (claudiodfc) [reply data/sql/updates/pending_db_world/rev_1623792334812044570.sql:3]: Also the readability is much better!

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**